packages/core/src/rules/oas2/remove-unused-components.ts

```diff
@@ -0,0 +1,55 @@
+import { Oas2Rule } from '../../visitors';
+import { Location } from '../../ref-utils';
+import { isEmptyObject } from '../../utils';
+
+export const RemoveUnusedComponents: Oas2Rule = () => {
+  let definitions = new Map<string, { used: boolean; name: string }>();
+
+  function registerComponent(location: Location, name: string): void {
+    definitions.set(location.absolutePointer, {
+      used: definitions.get(location.absolutePointer)?.used || false,
+      name,
+    });
+  }
+
+  return {
+    ref(ref, { type, resolve, key }) {
+      if (
+        ['Schema', 'Header', 'Parameter', 'Response', 'Example', 'RequestBody'].includes(type.name)
```

Comment:
This is invalid

oas2 supports only 

- Schema
- Parameter
- Response
- SecurityScheme